pkg/ipblock/ipblock.go

```diff
@@ -59,34 +66,40 @@ func (b *IPBlock) toIPRangesList() []string {
 
 // ContainedIn checks if this IP block is contained within another IP block.
 func (b *IPBlock) ContainedIn(other *IPBlock) bool {
+	if b == other {
+		return true
+	}
 	return b.ipRange.ContainedIn(other.ipRange)
 }
 
 // Intersect returns a new IPBlock from intersection of this IPBlock with input IPBlock
 func (b *IPBlock) Intersect(c *IPBlock) *IPBlock {
-	res := &IPBlock{}
-	res.ipRange = b.ipRange.Copy()
+	res := b.Copy()
+	if b == c {
+		return res
+	}
 	res.ipRange.Intersect(c.ipRange)
 	return res
 }
 
 // Equal returns true if this IPBlock equals the input IPBlock
 func (b *IPBlock) Equal(c *IPBlock) bool {
+	if b == c {
+		return true
+	}
 	return b.ipRange.Equal(c.ipRange)
 }
 
 // Subtract returns a new IPBlock from subtraction of input IPBlock from this IPBlock
 func (b *IPBlock) Subtract(c *IPBlock) *IPBlock {
-	res := &IPBlock{}
-	res.ipRange = b.ipRange.Copy()
+	res := b.Copy()
 	res.ipRange.Subtract(c.ipRange)
 	return res
 }
 
 // Union returns a new IPBlock from union of input IPBlock with this IPBlock
 func (b *IPBlock) Union(c *IPBlock) *IPBlock {
-	res := &IPBlock{}
-	res.ipRange = b.ipRange.Copy()
+	res := b.Copy()
```

Comment:
If aliasing is handled in `Intersect`, it should be handled in `Union` too (and maybe in `Subtract` as well).

Reply:
Missed it. Fixed.